qa-tests-backend/src/test/groovy/Services.groovy

```diff
@@ -204,6 +204,42 @@ class Services extends BaseService {
         return getViolationsHelper("Deployment:${deploymentName}+Policy:${policyName}", policyName, timeoutSeconds)
     }
 
+    static boolean checkForNoActiveViolations(
+            String deploymentName, String policyName, int timeoutSeconds) {
+        String query = "Deployment:${deploymentName}+Policy:${policyName}+Violation State:ACTIVE"
+
+        // Consecutive checks will reduce possibility that policy evaluation did not happened.
+        int consecutiveChecks = 3
+
+        int intervalSeconds = 3
+        int retries = (timeoutSeconds / intervalSeconds).intValue()
+        Timer t = new Timer(retries, intervalSeconds)
+        while (t.IsValid()) {
+            def violations = AlertService.getViolations(ListAlertsRequest.newBuilder()
```

Comment:
since we are doing it in a loop I think we should wrap it with try catch to retry if server returns error or timeout.